doc/requirements/sdid.rst

```diff
@@ -0,0 +1,261 @@
+
+There will always be a need for knowledgable users to communicate with however, as it is unreasonable to attempt to
+program in a fool-proof understanding of the physics coded for in the settings system. Such an endeavor would come with great developmental and maintenance costs.
+
+Most problems these days arise when users make typos or simple editing mistakes when manually adjusting some values in the case settings input files.
+
+Non-dynamic data typing
+^^^^^^^^^^^^^^^^^^^^^^^
+One specific category of safety in the settings system stems from the nature of data variables in Python.
+
+Because ARMI is written in Python, there's a large issue of dynamic typing polluting setting values. There can easily be aberrant behavior introduced by a new kind of data being input to a value.
+
+For example it's not uncommon to have code flow based off setting values such as zero. If some developer makes a mistake and changes the datatype of a value zero from an integer to a string, it will break the utility of the code flow evaluation without alerting anyone to the change in behavior.::
+
+	def myFunction(var):
+		if var == 0:
+	 	   doSomething()
+		else:
+		    doSomethingSubtlyDifferent()
+
+	myVar = 0
+	myFunction(myVar) # doSomething()
+
+	# oops! clumsy str() method used somewhere
+	myVar = "0"
+	myFunction(myVar) # doSomethingSubtlyDifferent()
+
+These kinds of hard to track mistakes were resolved by creating a more object oriented division of settings with type protection surrounding the setting's values, so it wouldn't be possible to set the example variable to a string.
+
+Now there are a finite set of allowable setting types which loosely mirror the Python primitive data types, namely: ``list``, ``float``, ``int``, ``str``, and ``bool``.
+
+As mutable objects are encompassed in the list of supported data types, it became important to prevent the perversion of their contained values with alterations that would bypass the protective methods on setting objects entirely. Such an example would be when a list is returned from the appropriate list setting object, and something is appended to it. This would bypass all error prevention methodology in place. The easiest fix for this was to return a deep copy of the contained value, and only reassign the contained value on an explicit value assignment statement::
+
+	# note 'cs' is the common term for 'case settings'
+	# referring to the settings system main shared object 'Settings'.
+	myList = cs['myListSetting']
+
+	myList.append('Some value not allowed by myListSetting!') # bad!
+	# if we stop the code here, the value in cs['myListSetting'] will not contain
+	# the bad value appended thanks to myList being a deep-copied value
+
+	# the following will run 'myList' against any error checking in the system
+	# either crashing the run or accepting the new value for later use.
+	cs['myListSetting'] = myList # good!
```

Comment:
I believe this type of assignment is not allowed anymore, right? We now have to use `modified()`, yes?